test/lsp4clj/io_chan_test.clj

```diff
@@ -50,6 +66,20 @@
                            "{\"key\":\"äpfel\"}"])
            (.toString output-stream "utf-8")))))
 
+(deftest output-stream-error-should-close-output-channel
+  (testing "when JSON serialisation fails"
+    (let [output-stream (java.io.ByteArrayOutputStream.)
+          output-ch (io-chan/output-stream->output-chan output-stream)]
+      (async/>!! output-ch {:not-serializable output-stream})
```

Comment:
suggestion: create a separate non-serializable object.

It appears that the `output-stream` is serving double duty here, both as part of creating the `output-ch`, but also as a non-serializable Object in the message. If that's so, it's clever but also opaque. I'd suggest creating a separate non-serializable thing to use in the message.

Reply:
Yeah, I was lazy there. Will change that.